word2vec.ipynb

```diff
@@ -0,0 +1,1140 @@
+{
```

Comment:
<p>These results are very interesting. Although cosine similarity for man and boy, and woman and girl, seem to become worse, the similar words list provides a different insight. </p><p> </p><p>By cosine similarity alone, it appears that the the first epoch is the best. However, when we look at the top 10 similar words to men, we see a strange list (I'm relying on Google Translate here):</p><p> </p><ol><li>Unknown word</li><li>Tell him</li><li>Unknown word</li><li>Unknown word</li><li>Boy</li><li>Unknown word</li><li>Each other</li><li>He grows</li><li>He loves her</li><li>He liked</li></ol><p> </p><p>But after 10 epochs this list is very different:</p><p> </p><ol><li>A woman</li><li>Girl</li><li>Man</li><li>Who</li><li>He is engaged</li><li>Love</li><li>Love</li><li>Boy</li><li>Is there</li><li>Named</li></ol><p> </p><p>To me, this looks like the 10 epoch result is converging on the concept of man as a person noun, similar to boy, woman, girl, etc, as well as relationships. Maybe the text about "Man" is dominated by biographies and other text about who they loved/were married to, etc. This definitely looks more meaningful than the results after 1 epoch. My guess as to why the cosine distance doesn't reveal much here is that to begin with, <em>all</em> words probably have a similar embedding. As the model trains, these words became more distant, but it is actually a sign of the model improving. To find convergence, maybe we should be training until the cosine similarities become relatively stable. We actually see possible signs of this around 10 epochs, where the tested cosine similarities are actually slightly higher than epoch 9. We would need more epochs to confirm.</p><p> </p><p>It could also help to see some results for non-gendered words. I know very little about Swahili, but the role of gender varies a great deal across cultures and languages. It would be interesting to see the comparison between other words, like "run" -&gt; "walk", "home" -&gt; "house", "fruit" -&gt; "food", etc.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/CallmeMehdi/swahili-word-embedding/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>